woocommerce/woocommerce-google-analytics-integration
# Fix inconsistencies across item data in events

### Changes proposed in this Pull Request:

This PR fixes inconsistencies across how data is structured for different events. Closes #200 

A `get_product_identifier` method is [introduced](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/d8ba3cee842fd45b3f854f19b989cc93a4e6813c/includes/class-wc-abstract-google-analytics-js.php#L116) and used for the item ID across [all events](https://github.com/woocommerce/woocommerce-google-analytics-integration/commit/0d6fdf7df9145956fde7e2d405cece393dd6b68f). The ID will either be the variation SKU, product SKU, or # + Product ID.

The [category line](https://github.com/woocommerce/woocommerce-google-analytics-integration/commit/ece7d75210502ed581967b0299521a29efcc9c88) is included with the add to cart event and the item name will always use the product title instead of the [variation name](https://github.com/woocommerce/woocommerce-google-analytics-integration/commit/fe67b9e69c9e6ea976979e880829fd6de16d7cc8) when an order is placed.

Additionally, Javascript is now used to build the [item ID and variant](https://github.com/woocommerce/woocommerce-google-analytics-integration/commit/81d30e57fd950dda15cbfa9c1cd656a36b797c76) when a product is added to cart to make sure the variation selection is considered.

### Detailed test instructions:

1. Ensure [Google Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) is installed and ON
2. Create a variable product
3. Add product to cart
4. View developer console and check the item ID, Name, Category, and Variant
5. Go to checkout
6. View developer console and check the item ID, Name, Category, and Variant. These should all match the values from step 4
8. Place order using any payment method
7. View developer console and check the item ID, Name, Category, and Variant. These should all match the values from step 4 and 6
